spec/std/http/client/client_spec.cr

```diff
@@ -65,6 +65,14 @@ module HTTP
         cl.port.should eq(9999)
       end
 
+      it "allows changing ssl settings" do
+        cl = Client.new(URI.parse("https://demo.com"))
+        cl.ssl_method = OpenSSL::SSL::Method::TLSv1
+        cl.ssl_cert_file = "/path/to/cert"
+        cl.ssl_key_file = "/path/to/key"
+        cl.ssl_ca_file = "/path/to/ca"
```

Comment:
One that allows passing an OpenSSL::SSL::Context to use for requests? I think that would make sense. I have that in my experimental docker lib: https://github.com/jeromegn/docker.cr/blob/master/src/core_ext/http/client.cr#L3-L5